geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/JsonPathParser.scala

```diff
@@ -128,6 +132,10 @@ private class JsonPathParser extends Parser {
 
   def Attribute: Rule1[PathAttribute] = rule { "." ~ oneOrMore(Character) ~> PathAttribute ~ !"()" }
 
+  def BracketedAttribute: Rule1[BracketedPathAttribute] = rule { ".['" ~ oneOrMore(CharacterWithDelimiter) ~> BracketedPathAttribute ~ !"()" ~ "']" }
+
+  def BracketedAttribute2: Rule1[BracketedPathAttribute] = rule { ".[" ~ oneOrMore(CharacterWithDelimiter) ~> BracketedPathAttribute ~ !"()" ~ "]" }
```

Comment:
it looks like not having quotes is invalid - we shouldn't match that case. Also, as currently written it would conflict with the `ArrayIndex` rule - requiring the quotes should fix that